api/handlers.go

```diff
@@ -2314,6 +2320,33 @@ func appendContentTypesFilterMods(mods *[]qm.QueryMod, f ContentTypesFilter) err
 	return nil
 }
 
+func appendDerivedTypesFilterMods(mods *[]qm.QueryMod, f DerivedTypesFilter) error {
+	if utils.IsEmpty(f.DerivedTypes) {
+		return nil
+	}
+
+	a := make([]interface{}, len(f.DerivedTypes))
+	for i, x := range f.DerivedTypes {
+		ct, ok := mdb.CONTENT_TYPE_REGISTRY.ByName[strings.ToUpper(x)]
+		if !ok {
+			return errors.Errorf("Unknown derive content type: %s", x)
+		}
+		if _, p := consts.PERMITTED_UNIT_CT_FOR_DERIVED_FILTER[ct.Name]; !p {
```

Comment:
maybe rename `p` to `ok`? 
as far as I know it does not conflict with the `ok` from above it as it is only declared in the `if` statement scope.

Reply:
fixed